### PR TITLE
Feat/dataset overview

### PR DESCRIPTION
**Pull request by lmacielvieira**
### Description

In order to improve user experience and better show the dataset specific information, a page that displays relevant dataset information was created. Some actions available on the dataset item card were removed and a link to navigate to the dataset overview page was added

###### Data set list page
- [x] Removed admin dataset actions from dataset item card on dataset list page and added a navigation to the Dataset overview page

###### Data set overview page
- [x] Created dataset overview page and route based on dataset id (dataset/:id)
- [x] Created dataset specific information area
- [x] Created dataset gallery viewer area (it is hidden for now)
- [x] Display the dataset title on information area
- [x] Display dataset submitter name on information area
- [x] Display dataset submitter group name and redirect user to group page, when the user user click on group name
- [x] Display dataset description on information area
- [x] Create an annotation component that shows dataset related annotations separated by database and fdr level
- [x] Redirect user to annotation page filtered by dataset/fdr/database when it applies. The applied scenario should happen when the user click on an annotation count on the annotation component or in the 'browse annotation' button
- [x] Display dataset metadata on information area
- [x] Display the dataset related projects on information area

###### Dataset description
- [x] Created dataset_description field on graphql.dataset table
- [x] Added  dataset_description to update, get and create dataset related queries

###### Dataset overview related
- [x] Added annotationCounts return field, containing the annotation counts of all ds item dbs
 
 
 
 #### Tests
 
 ##### Front end
 
- [x] Unit Test
- [ ] e2e Test
 
 ###### Browsers and resolutions
- [x] Chrome
- [ ]  Safari
- [x] sm, md, lg, xl, lg

image 
image
image
image
image

### CODE REVIEW

**Review by LachlanStuart — CHANGES_REQUESTED** [review]
Overall looks really good, and it's impressive that you got through this so quickly.

The feedback items I consider critical before merging:
* Renaming `datasetDescription` to `description` (it's very painful to rename API fields after release, so it's best to do it now if we do change it)
* Adding a feature flag so that the "Dataset overview" link is hidden by default
* Fixing the Datasets list quick filter
* Changing `MetadataEditor.spec.ts` to use `mount` instead of `shallowMount`

**Comment by LachlanStuart** [comment]
One more feedback item: I haven't added any specific feedback about the appearance of the Overview page as I feel we should wait until the diagnostics stuff is ready before doing a polishing pass to make it pretty. The main thing I think should be done is updating the typography (header & text styles) to match the new style of the content pages, e.g. [help](https://metaspace2020.eu/help) and [publications](https://metaspace2020.eu/publications). However, this can wait for a future PR.

**Review by LachlanStuart — APPROVED** [review]
The changes look good. You're free to merge this once the CI test/lint has been fixed

File: metaspace/webapp/src/modules/Datasets/list/DatasetItemActions.tsx [hunk 1]
```diff
@@ -233,6 +238,16 @@ const DatasetItemActions = defineComponent({
...
+
+          <div>
+            <i class="el-icon-data-analysis" />
+            <router-link to={{
+              name: 'dataset-overview',
+              params: { dataset_id: props.dataset.id },
+            }}>
+              Dataset overview
+            </router-link>
+          </div>
```
[the remark this answers is not in the document]
Comment (lmacielvieira) [reply metaspace/webapp/src/modules/Datasets/list/DatasetItemActions.tsx]: Added feat show_dataset_overview.
i.e http://xxxxx/datasets?feat=show_dataset_overview

File: metaspace/webapp/src/modules/Annotations/AnnotationsPage.vue [hunk 2]
```diff
@@ -1,7 +1,8 @@
 <template>
   <div id="annot-page">
```
[the remark this answers is not in the document]
Comment (lmacielvieira) [reply metaspace/webapp/src/modules/Annotations/AnnotationsPage.vue:2]: Added back button. (screenshot updated on PR description)

File: metaspace/webapp/src/modules/Datasets/overview/AnnotationCountTable.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      const { headerTitleSuffix } = props
+      return `${col === 'name' ? 'Database' : col}${col !== 'name' ? headerTitleSuffix : ''}`
+    }
+
+    const formatCell = (row: any, column: any, cellValue: any, rowIndex: number, colIndex: number) => {
+      const { id } = props
+      if (colIndex === 0) {
+        return (
+          <RouterLink to={annotationsLink(id?.toString(), row?.id,
+            undefined)}>
```
[the remark this answers is not in the document]
Comment (lmacielvieira) [reply metaspace/webapp/src/modules/Datasets/overview/AnnotationCountTable.tsx:87]: It was needed, as I removed the db in "defaultInLevels" for the specific case where annotation is accessed via dataset overview navigation.

**Pull request merged**